subsys/bluetooth/audio/unicast_client.c

```diff
@@ -1897,6 +1908,198 @@ static int unicast_client_ase_discover(struct bt_conn *conn,
 	return bt_gatt_read(conn, &params->read);
 }
 
+static uint8_t unicast_client_pacs_location_read_func(struct bt_conn *conn,
+						      uint8_t err,
+						      struct bt_gatt_read_params *read,
+						      const void *data,
+						      uint16_t length)
+{
+	struct bt_audio_discover_params *params;
+	struct net_buf_simple buf;
+	uint32_t location;
+
+	params = CONTAINER_OF(read, struct bt_audio_discover_params, read);
+
+	BT_DBG("conn %p err 0x%02x len %u", conn, err, length);
+
+	/* Location is an optional characteristic and may not exist; if not we
```

Comment:
I am not sure I follow.
If the location characteristic does not exist, the pacs_location_read will not be called in location_discover_cb, and as a result this function should not be called.
Is this just "to make sure"?

Reply:
Good point. In case that the read fails, we should actually abort